### ISSUE TITLE
Add validations before returning the Sandbox

### ISSUE THREAD

**Issue by xw19** [cri-o/cri-o#8822]
PR #8610 introduces a builder for initializing the pod sandbox. This means we build the pod sandbox step by step. In the end, we will request the builder to get the sandbox we have built.
When we request the builder to get the pod sandbox, the sandbox builder should validate that the required fields are present before returning the sandbox.

### Required fields:

    Id
    name
    createdAt
    config

**Comment by saschagrunert** [comment]
@xw19 thank you for opening the issue, do you consider to work on this?

**Comment by xw19** [comment]
Yes @saschagrunert

**Comment by xw19** [comment]
@haircommander @saschagrunert @kwilczynski @sohankunkerkar I believe the config and name are not required only, id, createdAt and criSandbox are required. Is the assumption correct?

**Comment by haircommander** [comment]
Eevery field we unconditionally set previously should be required IMO. I also wonder if there should be a function called counter to make sure we call all of the pieces of the construction. the source of anxiety motivating this is we miss some subtle field in the sandbox that formerly was passed to the sandbox constructor which would make it easier to catch

**Comment by xw19** [comment]
See this is an example of how we are using sbox.New earlier

```
mySandbox, err = sandbox.New(sandboxID, "", "", "", "",
make(map[string]string), make(map[string]string), "", "",
&types.PodSandboxMetadata{}, "", "", false, "", "", "",
[]*hostport.PortMapping{}, false, time.Now(), "", nil, nil)
```
This is how we are using now
```
// Setup test vars
createdAt := time.Now()
sbox := sandbox.NewBuilder()
sbox.SetID("sandboxID")
sbox.SetLogDir("test")
sbox.SetCreatedAt(createdAt)
err = sbox.SetCRISandbox(sbox.ID(), make(map[string]string), make(map[string]string), &types.PodSandboxMetadata{})
if err != nil {
panic(err)
}
sbox.SetPrivileged(false)
sbox.SetPortMappings([]*hostport.PortMapping{})
sbox.SetHostNetwork(false)
sbox.SetContainers(memorystore.New[*oci.Container]())
mySandbox, err = sbox.GetSandbox()
```

Most of these values are default type values. Not sure how we validate. Please explain a bit more

**Comment by xw19** [comment]
@haircommander can you check is this what you want for example if id is not given the Getsandbox should fail? https://github.com/cri-o/cri-o/pull/8908.

I feel the sandbox object currently is designed in a way that it will not fail even if fields like id is not there.

**Comment by haircommander** [comment]
yeah I almost wonder if we should have different functions for the unit tests vs actual production running. like if a pod doesn't have a number of those things it's not really functioning correctly, but we can make it through the unit tests without them

another way of thinking about it: what I really want is to ensure each of those setter methods are called, not necessarily requiring they're populated. It's a coding check to ensure on both load and create we call all the required methods. maybe instead of checking the value we ensure each function was called someway? what do @cri-o/cri-o-maintainers think

**Comment by xw19** [comment]
@haircommander  are you looking for something like this https://github.com/cri-o/cri-o/pull/8921

**Comment by xw19** [comment]
Completed #9008
